core/src/apps/solana/ui/format.py

```diff
@@ -0,0 +1,42 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ..transaction.instructions import Instruction
+
+
+def format_pubkey(_: Instruction, value: bytes | None) -> str:
+    from trezor.crypto import base58
+
+    if value is None:
+        raise ValueError  # should not be called with optional pubkey
+
+    return base58.encode(value)
+
+
+def format_lamports(_: Instruction, value: int) -> str:
+    from trezor.strings import format_amount
```

Comment:
small nit: `from trezor.strings import format_amount, format_timestamp` should be top-level import.

(the rules for function-local imports are somewhat unclear, but generally a single top-level import is preferred in case it replaces two or more function-local imports)

Reply:
https://github.com/trezor/trezor-firmware/pull/3359/commits/853af2f57fa017c5ae628344d4e1363c68936664#diff-0ed8070728ed70f5cd08bb7f7f5e4a53db0e6fff3002f21ca465408bfb9dc824R3